### ISSUE TITLE
Yarn install --offline for angular-cli app fails for @angular/common

### ISSUE THREAD

**Issue by vairamsvsjdo** [yarnpkg/yarn#2453]
0.18.1
**Do you want to request a *feature* or report a *bug*?**

Bug
**What is the current behavior?**
Yarn install --offline for angular-cli app fails for @angular/common
**If the current behavior is a bug, please provide the steps to reproduce.**

Have angular-cli angular2 all packages installed through npm and in npm cache
Connected to internet
git clone https://github.com/vairamsvsjdo/ng-yarn-sample app1
Cd app1
Yarn install
Cd..
git clone https://github.com/vairamsvsjdo/ng-yarn-sample app2
Disconnect from Internet
Cd app2
Yarn install
Receive error can't find any version's for "@angular/common" that matches "^2.3.1" in our cache
**What is the expected behavior?**
Be able to install all packages as defined in package.json for angular2 applications 
**Please mention your node.js, yarn and operating system version.**
Node v6.9.4
Yarn v0.18.1
Windows 10

**Comment by alexsaveliev** [comment]
Reproduced the same issue with node 6.9.4, yarn 0.19.1 on Ubuntu 15. It may affect many TypeScript 2.0+ projects with `@types/name` dependencies..

`package.json`:

```json
{
    "dependencies": 
    {
        "@types/diff": "0.0.31"
    }
}
```

log:

```
test@test-Lenovo-Y50-70:/tmp/yarn-bug$ yarn --cache-folder cache 
yarn install v0.19.1
info No lockfile found.
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 6.22s.
test@test-Lenovo-Y50-70:/tmp/yarn-bug$ rm -rf node_modules/ yarn.lock 
test@test-Lenovo-Y50-70:/tmp/yarn-bug$ yarn --cache-folder cache --offline
yarn install v0.19.1
info No lockfile found.
warning No license field
[1/4] Resolving packages...
error Couldn't find any versions for "@types/diff" that matches "0.0.31" in our cache. Possible versions: ""
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
test@test-Lenovo-Y50-70:/tmp/yarn-bug$ ls -al cache/
total 16
drwxrwxr-x 4 test test 4096 РЇРЅРІ 18 16:20 .
drwxrwxr-x 3 test test 4096 РЇРЅРІ 18 16:20 ..
drwxrwxr-x 3 test test 4096 РЇРЅРІ 18 16:20 npm-@types
drwxrwxr-x 2 test test 4096 РЇРЅРІ 18 16:20 .tmp
```

**Comment by rally25rs** [comment]
This was fixed by #2497 and confirmed fixed in v0.24.6.

Side-note, the original issue uses the `--cache-folder cache ` option. This would break the caching anyway, because `app1` would have an `app1/cache` dir where it's deps got downloaded and cached.

When you move to `app2` and try to `yarn --cache-folder cache` it looks for the cached files in `app2/cache` which does not exist, so it has to try to look online.
